### PR TITLE
fix: flaky test #NP-46162

### PR DESCRIPTION
**Pull request by torbjokv**

### CODE REVIEW

File: scopus-import/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java [hunk 1]
```diff
@@ -1754,21 +1754,13 @@ private Contributor findContributorByName(String givenName, List<Contributor> co
     }
 
     private void checkCollaborationName(CollaborationTp collaboration, List<Contributor> contributors) {
-        var optionalContributor = findContributorBySequence(collaboration.getSeq(), contributors);
-        if (Integer.parseInt(collaboration.getSeq()) < contributors.size()) {
-            var contributor = optionalContributor.getFirst();
+        var contributor = findContributorByName(collaboration.getIndexedName(), contributors);
 
-            assertEquals(1, optionalContributor.size());
+        if (Integer.parseInt(collaboration.getSeq()) < contributors.size()) {
```
[the remark this answers is not in the document]
Comment (torbjokv) [reply scopus-import/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java:1759]: me neither, remove the line? the test?

[the remark this answers is not in the document]
Comment (torbjokv) [reply scopus-import/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java]: I don't understood that line, so I removed it and test seems to still work.

File: scopus-import/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java [hunk 2]
```diff
@@ -1754,19 +1754,9 @@ private Contributor findContributorByName(String givenName, List<Contributor> co
...
-            assertEquals(1, optionalContributor.size());
-            assertEquals(collaboration.getIndexedName(), contributor.getIdentity().getName());
-        }
-    }
-
-    private List<Contributor> findContributorBySequence(String sequence, List<Contributor> contributors) {
-        return contributors.stream()
-                   .filter(contributor -> sequence.equals(String.valueOf(contributor.getSequence())))
-                   .collect(Collectors.toList());
+        assertEquals(collaboration.getIndexedName(), contributor.getIdentity().getName());
```
Comment (anetteOlli) [inline scopus-import/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java:1759]: If we already found the contributor by name it's a bit strange to assert that it has the same name. Change to assertThat(contributor, is(notNullValue())?

File: scopus-import/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java [hunk 3]
```diff
@@ -1754,19 +1754,9 @@ private Contributor findContributorByName(String givenName, List<Contributor> co
     }
 
     private void checkCollaborationName(CollaborationTp collaboration, List<Contributor> contributors) {
-        var optionalContributor = findContributorBySequence(collaboration.getSeq(), contributors);
-        if (Integer.parseInt(collaboration.getSeq()) < contributors.size()) {
-            var contributor = optionalContributor.getFirst();
+        var contributor = findContributorByName(collaboration.getIndexedName(), contributors);
```
Comment (anetteOlli) [inline scopus-import/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java:1757]: rename checkCollaborationName to checkThatCollaboration has been mapped.

Comment (torbjokv) [reply scopus-import/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java:1757]: or maybe just checkCollaboration

**Review by anetteOlli — APPROVED** [review]

**Review by brinxmat — APPROVED** [review]

**Pull request merged**